src/client/src/database.rs

```diff
@@ -283,55 +288,115 @@ impl Database {
+            // tonic error.
+            Err(status) => {
+                let e: error::Error = status.into();
+                let code = e.status_code();
+                let msg = e.to_string();
+                ServerSnafu { code, msg }.fail()
+            }
+            // decode error.
+            Ok(Err(err)) => Err(err),
+            Ok(Ok(first_flight_message_raw)) => match first_flight_message_raw {
+                FlightMessage::AffectedRows(rows) => {
+                    ensure!(
+                        flight_message_stream.next().await.is_none(),
+                        IllegalFlightMessagesSnafu {
+                            reason:
+                                "Expect 'AffectedRows' Flight messages to be the one and the only!"
+                        }
+                    );
+                    Ok(Output::AffectedRows(rows))
+                }
+                FlightMessage::Schema(schema) => {
+                    let stream: Pin<Box<dyn Stream<Item = RecordBatchResult<RecordBatch>> + Send>> =
+                        Box::pin(stream! {
+                            while let Some(flight_message) = flight_message_stream.next().await {
+                                match flight_message {
+                                    // tonic error.
+                                    Err(status) => {
+                                        let e: error::Error = status.into();
+                                        let code = e.status_code();
+                                        let msg = e.to_string();
+                                        yield ServerSnafu { code, msg }
+                                            .fail()
+                                            .map_err(BoxedError::new)
+                                            .context(ExternalSnafu);
+                                        break;
+                                    }
+                                    // decode error.
+                                    Ok(Err(err)) => {
+                                        yield Err(err)
+                                            .map_err(BoxedError::new)
+                                            .context(ExternalSnafu);
+                                        break;
+                                    }
+                                    Ok(Ok(FlightMessage::Recordbatch(record_batch))) => {
+                                        yield Ok(record_batch);
+                                    }
+                                    // unexpected flight message variants.
+                                    Ok(Ok(_)) => {
+                                        yield IllegalFlightMessagesSnafu {reason: "A Schema message must be succeeded exclusively by a set of RecordBatch messages"}
+                                            .fail()
+                                            .map_err(BoxedError::new)
+                                            .context(ExternalSnafu);
```

Comment:
`IllegalFlightMessagesSnafu` is already an client error, no need to "BoxedError" it